### ISSUE TITLE
sc.datasets.ebi_expression_atlas errors

### ISSUE THREAD

**Issue by fidelram** [scverse/scanpy#1082]
I wanted to use ` sc.datasets.ebi_expression_atlas` but  I get an error that seems to be related to code only valid in python 3.8. This error happens with python version `3.6.7`.

Code:
```python
adata = sc.datasets.ebi_expression_atlas("E-GEOD-98556")
```

```pytb
/scanpy/scanpy/datasets/_ebi_expression_atlas.py in download_experiment(accession)
     41
     42     _download(
---> 43         download_url + "experiment-design", experiment_dir / "experimental_design.tsv",
     44     )
     45     _download(

/scanpy/scanpy/readwrite.py in _download(url, path)
    877         except Exception:
    878             # Make sure file doesn’t exist half-downloaded
--> 879             path.unlink(missing_ok=True)
    880             raise
    881

TypeError: unlink() got an unexpected keyword argument 'missing_ok'

...
```
`missing_ok` was added to python v. 3.8 (https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink)

**Comment by fidelram** [comment]
Thanks!

**Comment by ivirshup** [comment]
Oops, didn't mean to close this initially.

The datasets still don't download right on master. I've opened #1102 to fix it.

**Comment by ivirshup** [comment]
Fixed by #1102
